### DISCUSSION TITLE
initialColorMode not working (create-react-app)

### DISCUSSION THREAD

**Discussion by TendaiChikwiri** [chakra-ui/chakra-ui#5051] [Q&A] [answered]
When I set initialColorMode to "dark", the site in dark mode does not load in dark mode

I followed the docs to add dark mode on my create-react-app site, the only difference between the documentation and my code is that initialColorMode is set to "dark"  instead of light as below

theme.js
``` 
import { extendTheme } from "@chakra-ui/react"

const config = {
  initialColorMode: "dark",
  useSystemColorMode: false,
}
const theme = extendTheme({ config })

export default theme
```
index.jsx
```import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { ColorModeScript } from "@chakra-ui/react";
import theme from "./theme";

ReactDOM.render(
  <React.StrictMode>
    <ColorModeScript initialColorMode={theme.config.initialColorMode} />
    <App />
  </React.StrictMode>,
  document.getElementById("root")
);
```

I'm not sure how to fix this, When I open the app in incognito browser and on my phone the "light" version is loaded. The only way for me to see the dark version is to toggle to the dark version.

Thank you.

**Comment by anubra266** [comment, marked as answer]
It's a know issue, it will be fixed in the next release

**Reply by TendaiChikwiri** [reply]
Okay, I see. Thank you.

**Comment by alexcraviotto** [comment]
I have the same problem... not solved yet?

**Reply by carlosbet** [reply]
It seems for now the problem persists: @chakra-ui/react": "2.1.2".

The only thing that this feature seems to do is setting the value of "chakra-ui-color-mode"  property in LocalStorage at the moment of start. So in React, you can simply do it 'by hand':

```
ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {localStorage.setItem('chakra-ui-color-mode', 'dark')}
    <ChakraProvider>
      <App />
    </ChakraProvider>
  </React.StrictMode>
);
```

I use this, and seems all works fine… 😅 😇

**Reply by ianchildress** [reply]
The only thing that works for me is @carlosbet solution. Thank you for that. I think this issue or similar has been ongoing since at least Dec '21.

**Reply by kousuke** [reply]
> This sets the default only, and prevent overrideing those set by user:
> 
> ```js
>   if (!localStorage.getItem("chakra-ui-color-mode")) {
>     localStorage.setItem("chakra-ui-color-mode", "dark");
>   }
> ```
For me, `localStorage.getItem("chakra-ui-color-mode")` returns "light" for first access....

**Reply by kousuke** [reply]
so I did like this
```typescript
if (!localStorage.getItem("chakra-ui-color-mode-default")) {
    localStorage.setItem("chakra-ui-color-mode", "dark")
    localStorage.setItem("chakra-ui-color-mode-default", "set")
}
```

**Comment by faizerturk** [comment]
Hi all, is this problem goes on @chakra-ui/react": "^2.5.1" ?  
I don't want to use dark theme but whenever I implement @chakra-ui/react its automatically become dark mode. When I open the inspect the root html tag like this : 

`<html lang="en" data-theme="dark" style="color-scheme: dark;">`

on the other hand my theme.ts file  : 
`const config: ThemeConfig = {
  initialColorMode: "light",
  useSystemColorMode: false,
};
`

and index.ts file is : 

`
...
        <ChakraProvider theme={theme}>
          <ColorModeScript initialColorMode={theme.config.initialColorMode} />
          <App />
        </ChakraProvider>
...
`

How can I fix this problem?

**Comment by tomimacia** [comment]
I run into this problem as well and this worked for me, at least for Next.js but it should work as well on Create React App. Just deleted de ColorModeProvider from _app.jsx, and setted the initialColorMode at ColorModeScipt on _document.jsx.

For next.js it looks like this:

```
import { Html, Head, Main, NextScript } from "next/document";
import { ColorModeScript } from "@chakra-ui/react";

export default function Document() {
  return (
    <Html lang="en">
      <Head />
      <body>
        <ColorModeScript initialColorMode='dark' />
        <Main />
        <NextScript />
      </body>
    </Html>
  );
}
```

For Create React App should look this:

```
// index.js

import { ColorModeScript } from '@chakra-ui/react'
import * as ReactDOM from 'react-dom/client'
import App from './App'
import theme from './theme'

const rootElement = document.getElementById('root')
ReactDOM.createRoot(rootElement).render(
  <>
    {/* 👇 Here's the script */}
    <ColorModeScript initialColorMode={'dark'} />
    <App />
  </>,
)
```
So far its working okay.
Source [https://chakra-ui.com/docs/styled-system/color-mode](url)

**Reply by kousuke** [reply]
Oh, next and react are different...

**Comment by hiroyukikumazawa** [comment]
I have been facing the same issue, so solved with following:

```tsx
ReactDOM.createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
    {localStorage.getItem('chakra-ui-color-mode')? <></>:<>{localStorage.setItem('chakra-ui-color-mode', 'dark')}</>}
    <Provider store={store}>
      <ChakraProvider theme={theme}>
        <App />
      </ChakraProvider>
    </Provider>
  </React.StrictMode>,
)
```

**Reply by evgeniishch** [reply]
Thx, very helpful. The issue still persists tho
